packages/components/src/custom-select-control-v2/legacy-component/test/index.tsx

```diff
@@ -87,7 +87,73 @@ describe.each( [
 	[ 'Uncontrolled', UncontrolledCustomSelectControl ],
 	[ 'Controlled', ControlledCustomSelectControl ],
 ] )( 'CustomSelectControl (%s)', ( ...modeAndComponent ) => {
-	const [ , Component ] = modeAndComponent;
+	const [ mode, Component ] = modeAndComponent;
+
+	it( 'Should select the first option when no explicit initial value is passed without firing onChange', async () => {
+		const mockOnChange = jest.fn();
+		render( <Component { ...legacyProps } onChange={ mockOnChange } /> );
+
+		expect(
+			screen.getByRole( 'combobox', {
+				expanded: false,
+			} )
+		).toHaveTextContent( 'violets' );
+
+		// Necessary to wait for onChange to potentially fire
+		await sleep();
+
+		expect( mockOnChange ).not.toHaveBeenCalled();
+	} );
+
+	it( 'Should pick the initially selected option if the value prop is passed without firing onChange', async () => {
+		const mockOnChange = jest.fn();
+		render(
+			<Component
+				{ ...legacyProps }
+				onChange={ mockOnChange }
+				value={ legacyProps.options[ 3 ] }
+			/>
+		);
+
+		expect(
+			screen.getByRole( 'combobox', {
+				expanded: false,
+			} )
+		).toHaveTextContent( 'amber' );
+
+		// Necessary to wait for onChange to potentially fire
+		await sleep();
+
+		expect( mockOnChange ).not.toHaveBeenCalled();
+	} );
+
+	// Using the "Uncontrolled" version of the component
+	// in order to apply controlled logic directly in the test
+	if ( mode === 'Uncontrolled' ) {
+		it( 'Should apply external controlled updates', async () => {
+			const { rerender } = render(
+				<Component
+					{ ...legacyProps }
+					value={ legacyProps.options[ 0 ] }
+				/>
+			);
+
+			const currentSelectedItem = screen.getByRole( 'combobox', {
+				expanded: false,
+			} );
+
+			expect( currentSelectedItem ).toHaveTextContent( 'violets' );
+
+			rerender(
+				<Component
+					{ ...legacyProps }
+					value={ legacyProps.options[ 1 ] }
+				/>
+			);
+
+			expect( currentSelectedItem ).toHaveTextContent( 'crimson clover' );
```

Comment:
Does it also make sense to verify that `onChange` was not called? Essentially like we do in the above tests:

```suggestion
			const mockOnChange = jest.fn();
			const { rerender } = render(
				<Component
					{ ...legacyProps }
					onChange={ mockOnChange }
					value={ legacyProps.options[ 0 ] }
				/>
			);

			const currentSelectedItem = screen.getByRole( 'combobox', {
				expanded: false,
			} );

			expect( currentSelectedItem ).toHaveTextContent( 'violets' );

			rerender(
				<Component
					{ ...legacyProps }
					onChange={ mockOnChange }
					value={ legacyProps.options[ 1 ] }
				/>
			);

			expect( currentSelectedItem ).toHaveTextContent( 'crimson clover' );

			await sleep();

			expect( mockOnChange ).not.toHaveBeenCalled();
```